include/mesh/boundary_info.h

```diff
@@ -974,6 +982,13 @@ class BoundaryInfo : public ParallelObject
   std::map<boundary_id_type, std::string> _es_id_to_name;
 };
 
+template <typename ModifyFunctor>
+void
+BoundaryInfo::modify_side(ModifyFunctor & modify_functor)
+{
+  modify_functor(_boundary_side_id);
+}
```

Comment:
This is kind of a weird API, isn't it?  It's no better encapsulated than a non-const getter function for _boundary_side_id would be, but it's more complicated syntax.

Reply:
Haha I was anticipating a response like this. I have no counter-argument. Are you open to a non-const getter? Otherwise I don't know how to do what I want to do in the stitch operation without making things pretty complicated

Reply:
Or actually I could add a pretty specific `BoundaryInfo::clear_stitched_boundary_ids(boundary_id_type bid_1, boundary_id_type bid_2)`

Reply:
Now that I've said it, the new method sounds most attractive to me...

Reply:
I'd be reasonably happy with that, except either name it ```clear_stitched_boundary_side_ids``` or have it work on node/edge/shellface ids too?

But sadly, I'd even be okay with the non-const getter right now.  The const getter we already have is enough to prevent us from making data structure changes without issues backwards compatibility.  I should have fought the camel's nose at that point, at this point most of the damage is already done.

Reply:
Even though strict encapsulation crimes have already been committed, I elected to go the more encapsulated route and added the `BoundaryInfo` method